src/main/java/org/dita/dost/reader/MergeTopicParser.java

```diff
@@ -202,15 +200,16 @@ public void startDocument() throws SAXException {
     @Override
     public void startElement(final String uri, final String localName, final String qName, final Attributes attributes)
             throws SAXException {
-        // Skip redundant <dita> tags.
-        if (ELEMENT_NAME_DITA.equals(qName)) {
-            rootLang = attributes.getValue(XML_NS_URI, "lang");
-            return;
-        }
         final AttributesImpl atts = new AttributesImpl(attributes);
         final String classValue = atts.getValue(ATTRIBUTE_NAME_CLASS);
 
-        if (TOPIC_TOPIC.matches(classValue)) {
+        if (ELEMENT_NAME_DITA.equals(qName)) {
+            rootLang = attributes.getValue(XML_NS_URI, "lang");
+            if (atts.getValue(ATTRIBUTE_NAME_ID) == null) {
+            	XMLUtils.addOrSetAttribute(atts, ATTRIBUTE_NAME_ID, GENERATED_DITA_ELEMENT_ID);
```

Comment:
Oh looking at the code again I see why this is confusing.

`<dita>` shouldn't have an ID (it's not legal in source). I double check that just in case it's added by an earlier preprocess step, but assuming it's not there, I add the generated `@id`.

The next portion of the code previously only worked with topics (which *must* have an ID), but I've added in support for `<dita>`:
```
if (TOPIC_TOPIC.matches(classValue) || ELEMENT_NAME_DITA.equals(qName)) {
            handleID(atts);
```

That `handleID` method assumes the original ID exists, which it does for topics, and now exists as that common token for `<dita>`. `handleID` moves `@id` to `@oid`, and is where we actually generate the unique-within-merged-context new `@id`.